backend-lucene/src/main/java/org/hibernate/search/v6poc/backend/lucene/search/reader/impl/MultiReaderFactory.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.v6poc.backend.lucene.search.reader.impl;
+
+import java.io.IOException;
+import java.lang.invoke.MethodHandles;
+import java.util.Set;
+
+import org.apache.lucene.index.IndexReader;
+import org.hibernate.search.v6poc.backend.lucene.index.spi.ReaderProvider;
+import org.hibernate.search.v6poc.backend.lucene.logging.impl.Log;
+import org.hibernate.search.v6poc.util.spi.LoggerFactory;
+
+/**
+ * Creates and closes the IndexReaders encompassing multiple indexes.
+ *
+ * @author Sanne Grinovero (C) 2011 Red Hat Inc.
+ */
+public final class MultiReaderFactory {
+
+	private static final Log log = LoggerFactory.make( Log.class, MethodHandles.lookup() );
+
+	private MultiReaderFactory() {
+		//not allowed
+	}
+
+	public static IndexReader openReader(Set<String> indexNames, Set<ReaderProvider> readerProviders) {
+		if ( readerProviders.size() == 0 ) {
+			return null;
+		}
+		else {
```

Comment:
I know it's pre-existing code, but maybe we should add a special case for when there's only one ReaderProvider?